From d1647dd90c5242eab4b8d1a62bfed7fc3b35868b Mon Sep 17 00:00:00 2001
Message-Id: <patch-v4-10.34-d1647dd90c5-20211117T101025Z-avarab@gmail.com>
In-Reply-To: <cover-v4-00.34-00000000000-20211117T101025Z-avarab@gmail.com>
References: <cover-v3-00.23-00000000000-20211116T114334Z-avarab@gmail.com>
	<cover-v4-00.34-00000000000-20211117T101025Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 25 Mar 2021 11:09:30 +0100
Subject: [PATCH v4 10/34] pack-objects doc: include relevant git-config(1)
 docs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Include a full listing of those configuration variables relevant to
pack-objects, this is possible because preceding commits selectively
peeled off the variable(s) in core.* and uploadpack.* into their own
files.

This is similar in spirit to my b6a8d09f6d8 (gc docs: include the
"gc.*" section from "config" in "gc", 2019-04-07).

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Documentation/git-pack-objects.txt | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/Documentation/git-pack-objects.txt b/Documentation/git-pack-objects.txt
index dbfd1f90175..87b1aebbab7 100644
--- a/Documentation/git-pack-objects.txt
+++ b/Documentation/git-pack-objects.txt
@@ -414,12 +414,18 @@ over user-wide config, and so forth).
 CONFIGURATION
 -------------
 
-Various configuration variables affect packing, see
-linkgit:git-config[1] (search for "pack" and "delta").
+Various configuration variables affect packing. Those variables and
+more are discussed in linkgit:git-config[1].
+
+The below documentation is selectively included from
+linkgit:git-config[1]. It should be all the configuration variables
+relevant to "pack-objects", but given how spread out they are when in
+doubt consult linkgit:git-config[1] for the full listing:
+
+include::config/core/bigFileThreshold.txt[]
+include::config/pack.txt[]
+include::config/uploadpack/pack-objects.txt[]
 
-Notably, delta compression is not used on objects larger than the
-`core.bigFileThreshold` configuration variable and on files with the
-attribute `delta` set to false.
 
 SEE ALSO
 --------
-- 
2.34.0.796.g126c2ced91c

